map/download-preplanned-map/src/main/java/com/esri/samples/download_preplanned_map/DownloadPreplannedMapController.java

```diff
@@ -0,0 +1,415 @@
+              areasOfInterestGraphicsOverlay.setVisible(false);
+
+              // show the result in the map view
+              mapView.setMap(downloadPreplannedOfflineMapResult.getOfflineMap());
+
+              // add the package to the list of opened map packages
+              if (!openedMobileMapPackages.contains(downloadPreplannedOfflineMapResult.getMobileMapPackage())) {
+                openedMobileMapPackages.add(downloadPreplannedOfflineMapResult.getMobileMapPackage());
+              }
+
+              // update the button text
+              downloadAreaButton.setText("View Downloaded Area");
+
+              // re-enable the UI
+              showWebMapButton.setDisable(false);
+              downloadAreaButton.setDisable(false);
+              deleteOfflineAreasButton.setDisable(false);
+
+              // display error details if the job fails
+            } else if (downloadPreplannedOfflineMapJob.getStatus() == Job.Status.FAILED) {
+              new Alert(Alert.AlertType.ERROR, "Download Preplanned Offline Map Job failed. Error: " + downloadPreplannedOfflineMapJob.getError());
+            }
+          });
+        } catch (InterruptedException | ExecutionException e) {
+          new Alert(Alert.AlertType.ERROR, "Could not create Default Parameters for the Download Preplanned Offline Map Job.").show();
+        }
+      });
+    } else {
+      new Alert(Alert.AlertType.ERROR, "No Preplanned Map Area selected for downloading.").show();
+    }
+  }
+
+  /**
+   * Checks for layer and table errors of an offline map result, and displays them if present.
+   *
+   * @param downloadPreplannedOfflineMapResult the result to query for errors
+   */
+  private void checkForOfflineMapResultErrors(DownloadPreplannedOfflineMapResult downloadPreplannedOfflineMapResult) {
+    if (downloadPreplannedOfflineMapResult.hasErrors()) {
+      // accumulate all layer and table errors into a single message
+      StringBuilder stringBuilder = new StringBuilder("Errors: ");
+
+      Map<Layer, ArcGISRuntimeException> layerErrors = downloadPreplannedOfflineMapResult.getLayerErrors();
+      layerErrors.forEach((layer, exception) ->
+              stringBuilder.append("Layer: ").append(layer.getName()).append(". Exception: ").append(exception.getMessage()).append(". ")
+      );
+
+      Map<FeatureTable, ArcGISRuntimeException> tableError = downloadPreplannedOfflineMapResult.getTableErrors();
+      tableError.forEach((table, exception) ->
+              stringBuilder.append("Table: ").append(table.getTableName()).append(". Exception: ").append(exception.getMessage()).append(". ")
+      );
+
+      // show the message
+      new Alert(Alert.AlertType.ERROR, "One or more errors occurred with the Offline Map Result: " + stringBuilder.toString()).show();
+    }
+  }
+
+  /**
+   * Resets the map view to show the web map from the portal.
+   */
+  @FXML
+  private void showWebMap() {
```

Comment:
I agree that these names are more specific, but since the methods get called at some places not as a result of a button click, I don't think it would be appropriate to add the buttons to the method name. e.g. `showWebMap()` and `closeAllGeoDatabases()` get called within `deleteOfflineAreas()`. The last one also gets called in `terminate()`.
That's semantics though, so happy to rename if you still think that's appropriate.